### PR TITLE
Auto Delete Custom Roles

### PR DESCRIPTION
**Pull request by Platformer444**
Automatically delete the custom role if you no longer meet the requirements to possess one.

### CODE REVIEW

File: modules/xp/weekly.ts [hunk 1]
```diff
@@ -140,6 +141,30 @@ export default async function getWeekly(nextWeeklyDate: Date) {
 		]);
 	}
 
+	const customRoles = (await config.guild.roles.fetch()).filter((role) =>
+		role.name.toLowerCase().includes("✨ "),
+	);
+	weeklyWinners.forEach(async (weeklyWinner) => {
+		await config.guild.members.fetch(weeklyWinner.user).then(async (guildMember) => {
+			const filtered = customRoles.filter((customRole) => {
```
Comment (RedGuy12) [inline modules/xp/weekly.ts:149]: move L123-125 of roles.ts to a function (search for PREFIX and replace other usages as well while you're at it)

in here, check if they qualify
if not, run that function and just delete the role

Comment (Platformer444) [reply modules/xp/weekly.ts]: maybe resolve it?

File: modules/xp/weekly.ts [hunk 2]
```diff
@@ -146,6 +147,21 @@ export default async function getWeekly(nextWeeklyDate: Date) {
 		]);
 	}
 
+	weeklyWinners.forEach(async (weeklyWinner) => {
+		await config.guild.members.fetch(weeklyWinner.user).then(async (guildMember) => {
+			if (await qualifiesForRole(guildMember)) return;
+
+			const customRole = getCustomRole(guildMember);
+
+			if (customRole === undefined) return;
```
Comment (RedGuy12) [inline modules/xp/weekly.ts:156]:
```suggestion
			if (!customRole) return;
```

File: modules/xp/weekly.ts [hunk 3]
```diff
@@ -146,6 +147,21 @@ export default async function getWeekly(nextWeeklyDate: Date) {
...
+
+			config.guild.roles.delete(
+				customRole,
+				"User no longer meets the requirements to have a Custom Role",
```
Comment (RedGuy12) [inline modules/xp/weekly.ts:160]:
```suggestion
				"No longer meets custom role requirements",
```

File: modules/xp/weekly.ts [hunk 4]
```diff
@@ -146,6 +147,21 @@ export default async function getWeekly(nextWeeklyDate: Date) {
...
 
+	weeklyWinners.forEach(async (weeklyWinner) => {
+		await config.guild.members.fetch(weeklyWinner.user).then(async (guildMember) => {
+			if (await qualifiesForRole(guildMember)) return;
+
+			const customRole = getCustomRole(guildMember);
+
+			if (customRole === undefined) return;
+
+			config.guild.roles.delete(
```
Comment (RedGuy12) [inline modules/xp/weekly.ts:158]:
```suggestion
			customRole.delete(
```

File: modules/xp/weekly.ts [hunk 5]
```diff
@@ -146,6 +147,21 @@ export default async function getWeekly(nextWeeklyDate: Date) {
...
+				customRole,
```
Comment (RedGuy12) [inline modules/xp/weekly.ts:159]:
```suggestion
```

**Review by RedGuy12 — COMMENTED** [review]
Some stylistic changes

File: modules/xp/weekly.ts (hunk 1 above)
Comment (RedGuy12) [reply modules/xp/weekly.ts]: > search for PREFIX and replace other usages as well while you're at it

Also, please revert the package.json changes. You will need to disable actions so they stop being recreated. In the future, don't use the main branch and you won't have a problem

After that, I can merge this

Comment (Platformer444) [reply modules/xp/weekly.ts]: Ah only if I was on my lappy

File: modules/roles.ts [hunk 6]
```diff
@@ -284,21 +279,15 @@ defineModal("customRole", async (interaction) => {
 });
 
 defineEvent("guildMemberRemove", async (member) => {
-	for (const [, role] of await config.guild.roles.fetch()) {
-		if (role.name.startsWith(PREFIX) && !role.members.size) {
-			await role.delete(`${member.user.tag} left the server`);
-		}
-	}
+	const role = getCustomRole(member as GuildMember);
```
[the remark this answers is not in the document]
Comment (Platformer444) [reply modules/roles.ts:282]: aight

**Review by RedGuy12 — APPROVED** [review]

**Pull request merged**